scraping/spiders/france_red_cross.py

```diff
@@ -0,0 +1,115 @@
+import re
+import logging
+
+import scrapy
+
+
+log = logging.getLogger(__name__)
+
+
+class FranceRedCrossSpider(scrapy.Spider):
+    """
+    Red Cross France crawler.
+
+    .. note::
+        Website seems to block access to some of its pages
+        from certain locations.
+        A way to tunnel the traffic through other locations
+        (Europe, U.S. seem to work well) may be needed.
+    """
+
+    name = 'france_red_cross'
+    allowed_domains = ['croix-rouge.fr']
+    start_urls = [
+        'https://www.croix-rouge.fr/Annuaire?structure_type%5B%5D=1'
+        '&structure_type%5B%5D=2&structure_type%5B%5D=3&structure_type%5B%5D=4'
+        '&field=-1&sector=-1&department=-1&address=&searchBtn=Rechercher',
+    ]
+
+    # There's 165 pages in this directory as of 2022-05-22.
+    # This here is to avoid potentially upsetting the sysops.
+    download_delay = 5
+
+    def parse(self, response, **kwargs):
+        blocks = response.css('article.item')
+        for block in blocks:
+            yield parse_point(block)
+
+        next_page = response.css('li.next a::attr(href)').get()
+        if next_page:
+            yield response.follow(next_page, callback=self.parse)
+
+
+def parse_point(block):
+    name = block.css('h2 a span::text').get().strip().title()
+
+    point = {
+        'name': name,
+        'country': 'France',
+        # XXX: oversimplified, need to understand if actions can be used here
+        'categories': 'Medical',
```

Comment:
By their categories do you mean options they have under "Filière" / "Secteur d'appartenance"? Or the "actions" listed under each location? Or is there something else I'm missing?

Whichever it is, I might need some help matching those, since it's not that obvious to me… thanks!

Reply:
Yes, the ones used in filters. Actions could be mapped secondarily and optionally. I’ll find the mapping for you by tomorrow

Reply:
I could derive these three as related and relevant:
```
    'DOMICILE': 'Accommodation',
    'ENFANTS & FAMILLES': 'Children',
    'PERSONNES HANDICAPEES': 'Medical',
```
And these through actions:
```
    'Aide Alimentaire': 'Food',
    'Aides financières': 'Finance',
    'Textile & Bric-à-Brac': 'Clothes',
    'Accès aux soins et bien être': 'Medical',
    'Accueil et orientation': 'Accommodation',
    'Accueil famille-enfant': 'Children',
    'Espace bébé parents': 'Children',
    'Point Hygiène': 'Medical',
    'Postes de secours': 'Medical',
```
What do you think?

Reply:
That's pretty good!

I suggest the following mapping:

```python
RED_CROSS_CATEGORIES_MAPPING = {
    "DOMICILE": "Accommodation",
    "ENFANTS & FAMILLES": "Children",
    "LUTTE CONTRE L'EXCLUSION": "Social Help",
    "PERSONNES HANDICAPEES": "Disability support",
    "SANITAIRE": "Medical"
}

RED_CROSS_ACTION_MAPPING = {
    "Aide Alimentaire": "Food",
    "Aides financières": "Finance",
    "Postes de secours": "Any Help",
    "Textile & Bric-à-Brac": "Clothes",
    "Dynamique jeunesse": "Children",
    "Accès aux soins et bien être": "Medical",
    "Accueil et orientation": "Accommodation",
    "Accueil famille-enfant": "Children",
    "Espace bébé parents": "Children",
    "Point Hygiène": "Medical",
}

DEFAULT_CATEGORY = "Any Help"
```

So, for example, if you get a point with category `SANITAIRE` and actions `Point Hygiène` + `Espace bébé parents` then the resulting categories will be `Medical,Children`.

Reply:
Cool, incorporated these as well!